lib/devise-security/models/secure_validatable.rb

```diff
@@ -30,19 +30,17 @@ def self.included(base)
             validation_condition = "#{login_attribute}_changed?".to_sym
 
             validates login_attribute, uniqueness: {
-                                          scope:          authentication_keys[1..-1],
-                                          case_sensitive: !!case_insensitive_keys
-                                        },
-                                        if: validation_condition
+                                         scope: authentication_keys[1..-1],
+                                         case_sensitive: !!case_insensitive_keys
+                                       },
+                                       if: validation_condition
 
             already_validated_email = login_attribute.to_s == 'email'
           end
 
           unless devise_validation_enabled?
             validates :email, presence: true, if: :email_required?
-            unless already_validated_email
-              validates :email, uniqueness: true, allow_blank: true, if: :email_changed? # check uniq for email ever
-            end
+            validates :email, uniqueness: true, allow_blank: true, if: :email_changed? unless already_validated_email || skip_email_uniqueness_validation
```

Comment:
I think this should be a method `validate_email_uniquenes?` similar to the `email_required?` method.